### ISSUE TITLE
Support P2P DMA

### ISSUE THREAD

**Issue by pogobanane** [vmuxIO/vmuxIO#62]
First, I boot the (nested guest) with iommu=pt. In this mode, the guest configures iommu as passthrough by default, which means a device can DMA to entire memory. so the qemu (host) side needs to configure IOMMU so that the pass-throughed device can DMA to the entire guest memory. And therefore libvfio sends VFIO_USER_DMA_MAP commands to map. Most address space is RAM, and in that case, libvfio sends VFIO_USER_DMA_MAP commands with a file descriptor that corresponds to the guest memory. However, some address space is not RAM but I/O spaces, which is managed by qemu. In this case, we get VFIO_USER_DMA_MAP with fd=-1. To properly handle DMA requests to that address region, vmux would need to request qemu (libvfio-user) to handle I/O. I think VFIO_USER_DMA_READ/WRITE are for it (https://github.com/nutanix/libvfio-user/blob/master/docs/vfio-user.rst#vfio_user_dma_write)
When we start vfio-e1000 program in the guest, VFIO first disables passthrough (this results in unmapping all current IOMMU settings, including I/O spaces) and then creates a new IOMMU map.
tldr; I think we can ignore the mapping request with fd==-1 for now. we probably need to handle them when we want to support P2P DMA
